Source/Orts.Formats.Msts/SignalScripts.cs

```diff
@@ -1660,201 +1657,153 @@ static public ArrayList process_FunctionCall(string FunctionStatement, IDictiona
+                        TraceError(linenumber, tokenType, TermString);
+                        //Debugprocess_FunctionCall(TermParts);
+                        return TermParts;
                     }
                 }
 
-                // try SIGFEAT definition
-
-                if (!termset)
+                tokenType = "SIGFEAT_"; // try SIGFEAT definition
+                if (TermString.StartsWith(tokenType))
                 {
-                    if (TermString.StartsWith("SIGFEAT_"))
+                    string partString = TermString.Substring(tokenType.Length);
+                    var sfIndex = SignalShape.SignalSubObj.SignalSubTypes.IndexOf(partString);
+                    if (sfIndex > -1)
                     {
-                        string partString = TermString.Substring(8);
-                        try
-                        {
-                            int sfIndex = Orts.Formats.Msts.SignalShape.SignalSubObj.SignalSubTypes.IndexOf(partString);
-                            TermParts = new SCRParameterType(SCRTermType.Sigfeat, sfIndex);
-                        }
-                        catch (Exception Ex)
-                        {
-                            Trace.TraceWarning("sigscr-file line {1} : Unknown SubType : {0}\nDetails {2}",
-                                partString, linenumber.ToString(), Ex.ToString());
-#if DEBUG_PRINT_IN
-                            File.AppendAllText(din_fileLoc + @"sigscr.txt", "Unknown SubType : " + partString + "\n");
-#endif
-                        }
-                        termset = true;
+                        //Debugprocess_FunctionCall(new SCRParameterType(SCRTermType.Sigfeat, sfIndex));
+                        return new SCRParameterType(SCRTermType.Sigfeat, sfIndex);
+                    }
+                    else // nothing found - set error
+                    {
+                        TraceError(linenumber, "parameter in statement", TermString);
                     }
                 }
+                //Debugprocess_FunctionCall(TermParts);
+                return TermParts;
+            }//process_TermPart
 
-                // nothing found - set error
+            private static void Debugprocess_FunctionCall(SCRParameterType TermParts)
+            {
+                Trace.WriteLine($"SignalScript: {TermParts.PartParameter}, {TermParts.PartType}");
+            }
 
-                if (!termset)
-                {
-                    Trace.TraceWarning("sigscr-file line {1} : Unknown parameter in statement : {0}", TermString, linenumber.ToString());
+            private static void TraceError(int lineNumber, string tokenType, string termString )
+            {
 #if DEBUG_PRINT_IN
-                    File.AppendAllText(din_fileLoc + @"sigscr.txt", "Unknown parameter : " + TermString + "\n");
+                                File.AppendAllText(din_fileLoc + @"sigscr.txt", $"Unknown {tokenType : " + termString + "\n");
```

Comment:
Doesn't this line want to be something like this?
```suggestion
                                File.AppendAllText(din_fileLoc + @"sigscr.txt", $"Unknown {tokenType} : {termString}\n");
```

Reply:
Peter,

Good catch. 
I was intending to remove the try .. catch but I don't remember making this sort of change along the way.

Doesn't look as though it would compile but I guess it was hidden by the #IF .. #ENDIF 

Best wishes,